core/protocol/all_services.go

```diff
@@ -208,7 +224,24 @@ func newServices(
 		svcs.eventForwarderEngine = evtforward.NewNoopEngine(svcs.log, svcs.conf.EvtForward)
 	}
 
-	// this is done to go around circular deps again...
+	svcs.oracle = oracles.NewEngine(svcs.log, svcs.conf.Oracles, svcs.timeService, svcs.broker)
+
+	svcs.ethCallEngine = ethcall.NewEngine(svcs.log, svcs.conf.EvtForward.EthCall, svcs.ethClient, svcs.eventForwarder)
+
+	if svcs.conf.IsValidator() {
+		go svcs.ethCallEngine.Start()
+	}
+
+	svcs.ethereumOraclesVerifier = oracles.NewEthereumOracleVerifier(svcs.log, svcs.witness, svcs.timeService, svcs.broker,
```

Comment:
Just occurred to me while I was writing the snapshot README: you need to register this engine with the snapshot engine so it can suck up the state. It happens a few lines below adding to it the to `svcs.snapshot.AddProviders()` call.

Reply:
i do not, good spot